project/geosource/models.py

```diff
@@ -76,7 +109,7 @@ class Source(PolymorphicModel, CeleryCallMethodsMixin):
 
     settings = models.JSONField(default=dict, blank=True)
     groups = models.ManyToManyField(Group, blank=True, related_name="geosources")
-    report = models.JSONField(default=dict, blank=True, encoder=DjangoJSONEncoder)
+    report = models.ForeignKey(SourceReporting, on_delete=models.PROTECT, null=True)
```

Comment:
report i nullable, so deleting report should set NULL on delete

Reply:
Yes correct, I put protected when I start the dev and forgot to change it.

Reply:
Done !